components/react/hooks/src/useScroll/index.js

```diff
@@ -0,0 +1,41 @@
+import {useState, useEffect} from 'react'
+
+export default function useScroll({axis = 'Y'} = {}) {
+  const getBoundingClientRect = () =>
+    window?.document ? document.body.getBoundingClientRect() : 0
+
+  const AXIS_MAP = {
+    Y: 'top',
+    X: 'left'
+  }
+  const boundingKey = AXIS_MAP[axis]
+
+  const [scroll, setScroll] = useState({
+    bodyOffset: getBoundingClientRect(),
+    position: getBoundingClientRect()[boundingKey],
+    scrollDirection: null
+  })
+
+  const onScroll = e => {
+    const currentBounding = getBoundingClientRect()
+    const currentPosition = -currentBounding[boundingKey]
+
+    setScroll(({position: prevPosition}) => ({
+      bodyOffset: currentBounding,
+      position: currentPosition,
+      scrollDirection: prevPosition > currentPosition ? 'DOWN' : 'UP'
+    }))
+  }
+
+  useEffect(() => {
```

Comment:
We could use `useMount` to avoid having the empty dependency with the disable of the linter.